app/notifications/validators.py

```diff
@@ -40,31 +42,67 @@ def check_service_over_api_rate_limit(service, api_key):
 def check_service_over_daily_message_limit(key_type, service):
     if key_type != KEY_TYPE_TEST and current_app.config['REDIS_ENABLED']:
         cache_key = daily_limit_cache_key(service.id)
-        service_stats = redis_store.get(cache_key)
-        if not service_stats:
-            service_stats = services_dao.fetch_todays_total_message_count(service.id)
-            redis_store.set(cache_key, service_stats, ex=3600)
-
-        if (service.message_limit - int(service_stats) <= 100):
-            current_app.logger.info('service {} nearing daily limit {} - {}'.format(
-                service.id,
-                service.message_limit,
-                service_stats
-            ))
-
-        if int(service_stats) >= service.message_limit:
-            current_app.logger.info(
-                "service {} has been rate limited for daily use sent {} limit {}".format(
-                    service.id, int(service_stats), service.message_limit)
-            )
-            raise TooManyRequestsError(service.message_limit)
+        messages_sent = redis_store.get(cache_key)
+        if not messages_sent:
+            messages_sent = services_dao.fetch_todays_total_message_count(service.id)
+            redis_store.set(cache_key, messages_sent, ex=3600)
+
+        warn_about_daily_message_limit(service, int(messages_sent))
 
 
 def check_rate_limiting(service, api_key):
     check_service_over_api_rate_limit(service, api_key)
     check_service_over_daily_message_limit(api_key.key_type, service)
 
 
+def warn_about_daily_message_limit(service, messages_sent):
+    nearing_daily_message_limit = messages_sent >= .8 * service.message_limit
+    over_daily_message_limit = messages_sent >= service.message_limit
+
+    current_time = datetime.utcnow().isoformat()
+    cache_expiration = int(timedelta(days=1).total_seconds())
```

Comment:
I think in a larger, high traffic system we should time bound this to the end of the day, but I think it's ok for today.